server/etcdmain/config.go

```diff
@@ -97,17 +97,28 @@ func newConfig() *config {
 			string(cconfig.V2_DEPR_1_WRITE_ONLY_DROP),
 			string(cconfig.V2_DEPR_2_GONE)),
 	}
-
 	fs := cfg.cf.flagSet
 	fs.Usage = func() {
 		fmt.Fprintln(os.Stderr, usageline)
 	}
-
+	AddEmbedFlags(&cfg.ec, fs)
 	fs.StringVar(&cfg.configFile, "config-file", "", "Path to the server configuration file. Note that if a configuration file is provided, other command line flags and environment variables will be ignored.")
+	fs.Var(cfg.cf.fallback, "discovery-fallback", fmt.Sprintf("Valid values include %q", cfg.cf.fallback.Valids()))
+	fs.Var(cfg.cf.clusterState, "initial-cluster-state", "Initial cluster state ('new' when bootstrapping a new cluster or 'existing' when adding new members to an existing cluster). After successful initialization (bootstrapping or adding), flag is ignored on restarts.")
+	fs.Var(cfg.cf.v2deprecation, "v2-deprecation", fmt.Sprintf("v2store deprecation stage: %q. ", cfg.cf.v2deprecation.Valids()))
+
+	fs.BoolVar(&cfg.printVersion, "version", false, "Print the version and exit.")
+	// ignored
+	for _, f := range cfg.ignored {
+		fs.Var(&flags.IgnoredFlag{Name: f}, f, "")
+	}
+	return cfg
+}
 
+func AddEmbedFlags(cfg *embed.Config, fs *flag.FlagSet) {
```

Comment:
Add a method `(*embed.Config) AddFlags(fs *flag.FlagSet)` under the `server/embed `package, so as to encapsulate the details in the `embed` package?

Reply:
Right, I decided to first separate this step to make the review simpler. 
I can add a separate commit with your suggestion.

Reply:
Done